shell/common/persistent_cache.h

```diff
@@ -49,25 +49,48 @@ class PersistentCache : public GrContextOptions::PersistentCache {
   bool IsDumpingSkp() const { return is_dumping_skp_; }
   void SetIsDumpingSkp(bool value) { is_dumping_skp_ = value; }
 
+  // |GrContextOptions::PersistentCache|
+  sk_sp<SkData> load(const SkData& key) override;
+
+  using SkSLCache = std::pair<sk_sp<SkData>, sk_sp<SkData>>;
+
+  /// Load all the SkSL shader caches in the right directory.
+  std::vector<SkSLCache> LoadSkSLs();
+
+  static bool cache_sksl() { return cache_sksl_; }
+  static void SetCacheSkSL(bool value);
+  static void MarkStrategySet() { strategy_set_ = true; }
+
```

Comment:
I wonder if we can use some builder like pattern here. Like:

```
PersistentCacheBuilder
  .setCacheSkSl(true)
  ...
  .build()
```

That way we could avoid having this extra flag. But i'm not sure if this is a common c++ paradigm.

Reply:
We use this pattern in the `embedder_unittests` in for the `EmbedderConfigBuilder`.

Reply:
I like the builder pattern, but it doesn't seem to be compatible with our current ` PersistentCache::GetCacheForProcess` usages. If we think that it's worth to do a yak shave to remove ` PersistentCache::GetCacheForProcess` and let `Shell` own the `PersistentCache` used by `Rasterizer` and `IOManager`, maybe we shall do that in another isolated PR and create an issue to track its progress.